TypeSystem/flow-analysis/reachability_do_while_A03_t01.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion do while statement: If `N` is a do while statement of the form
+/// `do S while (E)` then:
+/// - Let `before(S) = conservativeJoin(before(N), assignedIn(N), capturedIn(N))`.
+/// - Let `before(E) = join(after(S), continue(N))`
+/// - Let `after(N) = join(false(E), break(S))`
+///
+/// @description Checks that `after(N) = join(false(E), break(S))`. Test that if
+/// `S` throws then an assignment in `S` is ignored by flow analysis.
```

Comment:
That's again slightly dangerous to say: For example, we could make the flow analysis "unignore" the assignment if we put a try/catch statement around the entire loop. It sounds perhaps a bit too magic if we can ignore something and later "unignore" it.

To keep the description closer to the actual rules, perhaps we could say that `n;` makes `after(N)` unreachable, which implies that the assignedness of `i` after the loop is the assignedness of `i` after having taken the "else" branch of the `if` statement.